txclib/project.py

```diff
@@ -1332,7 +1319,9 @@ def _get_url_by_pull_mode(self, mode):
         url = None
         if mode is not None:
             try:
-                url = PULL_MODE_URL_MAPPING[mode]
+                # check if specific url name exists in the url list
+                API_URLS[PULL_MODE_URL_NAME.format(mode=mode)]
+                url = PULL_MODE_URL_NAME.format(mode=mode)
```

Comment:
I think the following is cleaner:
```        if mode is not None:
            try:
                # check if specific url name exists in the url list
                API_URLS[PULL_MODE_URL_NAME.format(mode=mode)]
                return PULL_MODE_URL_NAME.format(mode=mode)
            except KeyError:
                logger.warning('Invalid mode provided. ' +
                               'Default pull mode will be used')
                return DEFAULT_PULL_URL```

I also don't like that `+` for string formatting, but I guess we can let it be, since it is already there.